ci/svd2rust-regress/src/github.rs

```diff
@@ -93,7 +93,7 @@ fn find_executable(dir: &Path, begins: &str) -> Result<Option<PathBuf>, anyhow::
                     .path()
                     .extension()
                     .is_some_and(|s| s == std::env::consts::EXE_EXTENSION))
-            && entry.path().extension().is_none_or(|s| s != "gz")
+            && entry.path().extension().map_or(true, |s| s != "gz")
```

Comment:
why this change?

Reply:
this API is not available in the MSRV version of Rust. But I also need to investigate why this wasn´t an issue with the old CI

Reply:
I think what's required here is that the regress tool still needs to be built with stable (or at least v1.82), and then only use v1.76 for checking the crate. I think this will be difficult with the matrix options. However, what we might be able to do (and I already did some prep-work for this) is that this MSRV rust version is specified in a toolchain TOML file in the generated project.